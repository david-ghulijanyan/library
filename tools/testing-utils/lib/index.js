import * as jestDom from "@testing-library/jest-dom";
import * as reactHooks from "@testing-library/react-hooks";
import * as userEvent from "@testing-library/user-event";
import * as api from "./api";
// import * as data from "./data";
import * as mocks from "./mocks";
import * as spies from "./spies";
import * as utils from "./utils";

const { react } = utils;

const def = {
	api,
	// data,
	mocks,
	spies,
	...utils,
	jestDom,
	reactHooks,
	userEvent,
};

export default def;

export { api, mocks, spies, react, jestDom, reactHooks, userEvent };
