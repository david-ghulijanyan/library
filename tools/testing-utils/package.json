{
	"name": "@library/testing-utils",
	"version": "0.1.0",
	"description": "Testing Utils",
	"main": "dist/cjs/index.js",
	"module": "dist/esm/index.js",
	"private": false,
	"author": {
		"email": "david.hayki.ghulijanyan@gmail.com",
		"name": "David Ghulijanyan"
	},
	"license": "UNLICENSED",
	"engines": {
		"node": ">=16.13.0",
		"npm": ">=8.1.0"
	},
	"files": [
		"dist"
	],
	"publishConfig": {
		"registry": "http://localhost:4873"
	},
	"scripts": {
		"build": "rimraf dist && npm run build:esm && npm run build:cjs",
		"postbuild2": "npm run clean:dist",
		"build:esm": "cross-env NODE_ENV=production BABEL_ENV=esm babel ./lib --copy-files --out-dir 'dist/esm' --source-maps --ignore 'node_modules,**/node_modules'",
		"build:cjs": "cross-env NODE_ENV=production BABEL_ENV=cjs babel ./lib --copy-files --out-dir 'dist/cjs' --source-maps --ignore 'node_modules,**/node_modules'",
		"clean:dist": "rimraf $(find ./dist -regextype egrep -regex \".*/__snapshots__|.*/__tests__|.*.spec.js$|.*.test.js$\")",
		"start": "node scripts/start.js",
		"test": "node scripts/test.js"
	},
	"eslintConfig": {
		"extends": [
			"@library/eslint-config"
		]
	},
	"prettier": "@library/prettier-config",
	"browserslist": {
		"production": [
			">0.2%",
			"not dead",
			"not op_mini all"
		],
		"development": [
			"last 1 chrome version",
			"last 1 firefox version",
			"last 1 safari version"
		]
	},
	"peerDependencies": {
		"axios": "^0.23.0",
		"react": "^17.0.2",
		"react-dom": "^17.0.2"
	},
	"dependencies": {
		"@testing-library/jest-dom": "^5.14.1",
		"@testing-library/react": "^12.1.2",
		"@testing-library/react-hooks": "^7.0.2",
		"@testing-library/user-event": "^13.5.0",
		"babel-jest": "^27.3.1",
		"faker": "^5.5.3",
		"jest": "^27.3.1",
		"jest-circus": "^27.3.1",
		"jest-cli": "^27.3.1",
		"jest-prop-type-error": "^1.1.0",
		"jest-resolve": "^27.3.1",
		"jest-watch-typeahead": "^1.0.0",
		"msw": "^0.35.0"
	},
	"devDependencies": {
		"@babel/cli": "^7.15.7",
		"@babel/core": "^7.15.8",
		"@library/babel-preset": "file:../../tools/babel-preset",
		"@library/eslint-config": "file:../../tools/eslint-config",
		"@library/prettier-config": "file:../../tools/prettier-config",
		"cross-env": "^7.0.3",
		"rimraf": "^3.0.2"
	}
}
