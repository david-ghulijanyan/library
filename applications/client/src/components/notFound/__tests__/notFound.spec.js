import { react as utils } from "@library/testing-utils";
// import * as router from "react-router-dom";
import NotFound from "..";

jest.mock("react-router-dom", () => ({
	useHistory: () => ({
		goBack: jest.fn(),
	}),
}));

const { render } = utils;

describe("NotFound component", () => {
	test("Should render properly with 404 error code", () => {
		// Arrange
		// Act
		const { container } = render(NotFound);

		// Assert
		expect(container).toHaveTextContent(/Error: 404 Page not found! Please, go back or go to home page/);
		expect(container).toMatchSnapshot();
	});
	test("Should render properly with 500 error code", () => {
		// Arrange
		// Act
		const { container } = render(NotFound, { code: 500, message: "Internal server error" });

		// Assert
		expect(container).toHaveTextContent(/Error: 500 Internal server error Please, go back or go to home page/);
		expect(container).toMatchSnapshot();
	});
});
