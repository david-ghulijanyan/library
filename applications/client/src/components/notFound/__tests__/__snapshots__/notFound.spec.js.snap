// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component Should render properly with 404 error code 1`] = `
<div>
  <div
    class="error"
  >
    <h1>
      Error: 
    </h1>
    <p
      class="error-code"
    >
      404
       
    </p>
    <p
      class="error-message"
    >
      Page not found!
       
    </p>
    <div
      class="error-actions"
    >
      <span>
        Please, 
      </span>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        go back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <span>
         or 
      </span>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        go to home page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NotFound component Should render properly with 500 error code 1`] = `
<div>
  <div
    class="error"
  >
    <h1>
      Error: 
    </h1>
    <p
      class="error-code"
    >
      500
       
    </p>
    <p
      class="error-message"
    >
      Internal server error
       
    </p>
    <div
      class="error-actions"
    >
      <span>
        Please, 
      </span>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        go back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <span>
         or 
      </span>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        go to home page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
